virttest/qemu_monitor.py

```diff
@@ -2713,18 +2713,58 @@ def operate_dirty_bitmap(self, operation, node, name, granularity=65536):
         """
         Add, remove or clear a dirty bitmap.
 
+        NOTE: this method is deprecated, please use block_dirty_bitmap_add,
+              block_dirty_bitmap_remove, block_dirty_bitmap_clear instead.
+
         :param operation: operations to bitmap, can be 'add', 'remove', and 'clear'
         :param node: device/node on which to operate dirty bitmap
         :param name: name of the dirty bitmap to operate
         :param granularity: granularity to track writes with
         """
+        return self._operate_dirty_bitmap(operation, node, name,
+                                          granularity=granularity)
+
+    def _operate_dirty_bitmap(self, operation, node, name, **kargs):
+        """
+        Add, remove or clear a dirty bitmap.
+
+        :param operation: operations to bitmap, can be 'add', 'remove', or
+                          'clear'
+        :param node: device/node on which to operate dirty bitmap
+        :param name: name of the dirty bitmap to operate
+        """
         cmd = "block-dirty-bitmap-%s" % operation
         self.verify_supported_cmd(cmd)
         args = {"node": node, "name": name}
-        if operation == "add":
-            args["granularity"] = granularity
+        args.update(kargs)
         return self.cmd(cmd, args)
 
+    def block_dirty_bitmap_add(self, node, name, granularity=None,
+                               persistent=None):
+        """
+        Add a dirty bitmap.
+
+        :param node: device node
+        :param name: bitmap name
+        :param granularity: segment size
+        :param persistent: persistent through QEMU shutdown
+        """
+        return self._operate_dirty_bitmap("add",
+                                          **{k: v for k, v in locals().items()
+                                             if k != "self" and v is not None})
```

Comment:
Request to pass the arguments directly instead of doing the iteration like this.